src/lib.rs

```diff
@@ -149,3 +149,41 @@ pub fn reflink_or_copy(from: impl AsRef<Path>, to: impl AsRef<Path>) -> io::Resu
 
     inner(from.as_ref(), to.as_ref())
 }
+/// Checks whether reflink is supported on the filesystem for the specified source and target paths.
+///
+/// This function verifies that both paths are on the same volume and that the filesystem supports
+/// reflink.
+///
+/// > Note: Currently the function works only for windows. It returns `Ok(ReflinkSupport::Unknown)`
+/// > for any other platform.
+///
+/// # Example
+/// ```
+/// fn main() -> std::io::Result<()> {
+///     let support = reflink_copy::check_reflink_support("C:\\path\\to\\file", "C:\\path\\to\\another_file")?;
+///     println!("{support:?}");
+///     let support = reflink_copy::check_reflink_support("path\\to\\folder", "path\\to\\another_folder")?;
+///     println!("{support:?}");
+///     Ok(())
+/// }
+/// ```
+pub fn check_reflink_support(
+    from: impl AsRef<Path>,
+    to: impl AsRef<Path>,
+) -> io::Result<ReflinkSupport> {
+    #[cfg(windows)]
```

Comment:
We could do a `.is_file()` check here, just like `reflink`

Reply:
I actually want this function to work with folders. The general case that I had in mind is next: a program needs to copy a folder to another location. The program calls this function once with the source and target folders. Depending on the result, it will either copy or clone the files.

Calling the function once, instead of for each file, makes the process faster and more efficient.

Reply:
Hmm, in that case, I think you'd need to update the `reflink` function as well.

Currently it rejects any non-file immediately

Reply:
I think it's fine, I've added `copy_folder` example to demonstrate the use case

Reply:
oh you are right sorry, we only check it on error

https://github.com/cargo-bins/reflink-copy/blob/9e0b2cf4278146c19782ea458a73a39871d21dd5/src/lib.rs#L75